### PR TITLE
♻️Move amp-consent to toggle compatible implementation

### PR DESCRIPTION
**Pull request by jridgewell**
Part of #17475

### CODE REVIEW

File: extensions/amp-consent/0.1/amp-consent.js [hunk 1]
```diff
@@ -277,7 +279,7 @@ export class AmpConsent extends AMP.BaseElement {
       // Display the current instance
       this.currentDisplayInstance_ = instanceId;
       const uiElement = this.consentUI_[this.currentDisplayInstance_];
-      setImportantStyles(uiElement, {display: 'block'});
```
Comment (jridgewell) [inline extensions/amp-consent/0.1/amp-consent.js]: I'm assuming this was just to undo the `display: none`, correct? You don't mean to actually force `display: block`?

File: extensions/amp-consent/0.1/amp-consent.js [hunk 2]
```diff
@@ -673,8 +672,7 @@ export class AmpConsent extends AMP.BaseElement {
         classList.add('amp-active');
         classList.remove('amp-hidden');
         this.getViewport().addToFixedLayer(this.element);
-        setImportantStyles(dev().assertElement(this.postPromptUI_),
-            {display: 'block'});
+        toggle(dev().assertElement(this.postPromptUI_), true);
```
Comment (jridgewell) [inline extensions/amp-consent/0.1/amp-consent.js]: Again, assuming you just mean to undo `display: none`

File: extensions/amp-consent/0.1/amp-consent.css [hunk 3]
```diff
@@ -23,14 +23,13 @@
...
   width: 100%;
- }
+}
 
- amp-consent > * {
-   display: none !important;
-   max-height: 100vh !important;
- }
+amp-consent > * {
+  max-height: 100vh !important;
```
[the remark this answers is not in the document]
Comment (zhouyx) [reply extensions/amp-consent/0.1/amp-consent.css]: Would there be a gap before `<amp-consent>` buildCallback?

[the remark this answers is not in the document]
Comment (jridgewell) [reply extensions/amp-consent/0.1/amp-consent.css]: Added an additional rule to ensure that won't happen.

**Comment by jridgewell** [comment]
> Did we change the implementation for toggle already?

Not yet. Will be coming shortly.

> Otherwise we will still face the problem with display : none !important

There's no `display : none !important` to contend with. https://github.com/ampproject/amphtml/pull/17819/files#diff-5a11a42c72b4420f5141119de84337d7R29 will fail to match as soon as the element is built.

**Pull request merged**